docs/config.md

```diff
@@ -0,0 +1,37 @@
+# Configuration
+
+The CSI driver instance is configured by a combination of command-line flags and a YAML configuration
+file. You can use the following flags:
+
+- `--config`: path to the YAML configuration file, optional
+- `--nodeid`: the current Kubernetes node identifier, must be the same than the node it's running on
+- `--endpoint`: CSI driver API endpoint for Kubernetes kubelet
+- `--drivername`: CSI driver name to be registered in the cluster
+- `--maxvolumespernode`: maximum amount of volumes per node
+
+During the rollout of the CSI driver it captures the node-id via `env.ValueFrom`
+[directive](./deploy/csi-hostpath-plugin.yaml).
+
+## Configuration File
+
+By default the CSI driver expects the configuration at `/var/run/configmaps/config/config.yaml`, and
+the following attributes:
+
+```yml
+---
+# interval to relist all "Share" object instances
+shareRelistInterval: 10m
+
+# toggles actively watching for resources, when disabled it will only read objects before mount
+refreshResources: true
+
+# list of namespace names ignored
+ignoredNamespaces: []
+```
+
+When the file is not present, the driver assumes default values instead. And, when the configuration
+contents change,  it restarts after a couple second, allowing Kubernetes to restart it back again,
+with updated configs.
+
+During the rollout, a `config` named `ConfigMap` is created with default configuration values, in the
```

Comment:
you mean `ConfigMap` named `config` I think

that said, let's get a little more precise in our name, as we'll be sharing a namespace with the other CSI drivers

go with `csi-driver-shared-resource-config`